graph/edge/impl/ThingEdgeImpl.java

```diff
@@ -220,6 +222,87 @@ public final int hashCode() {
         }
     }
 
+    public static class Virtual extends ThingEdgeImpl implements ThingEdge {
```

Comment:
we create a virtual edge that may not exist when we want to use a `seek` operation on an edge iterator! Pretty interesting concept!